packages/flutter/lib/src/widgets/selection_container.dart

```diff
@@ -179,7 +179,21 @@ class _SelectionContainerState extends State<SelectionContainer> with Selectable
   @override
   SelectedContent? getSelectedContent() {
     assert(!widget._disabled);
-    return widget.delegate!.getSelectedContent();
+    final SelectedContent? content = widget.delegate!.getSelectedContent();
+    if (content == null) return null;
+
+    final String contentStr = content!.plainText;
+
+    // If the content string is empty or already ends with a newline character,
```

Comment:
```suggestion
    // If the content string is empty or already ends with a newline character,
    // then return the original SelectedContent as-is.
    if (contentStr.isEmpty || contentStr.endsWith('\n')) return content;
```

Use an early return on isEmpty & use the `endsWith()` method instead.